src/allomancy/java/leaf/cosmere/allomancy/common/manifestation/AllomancyIronSteel.java

```diff
@@ -541,4 +574,220 @@ public static void CheckRecipeForMetal(TagKey<Item> containsMetal, Recipe<?> rec
 			}
 		}
 	}
+
+	// set scan range of Lines Thread
+	public static void setScanRange(int range)
+	{
+		linesThread.setScanRange(range);
+	}
+
+	// this should be threaded to avoid lag spikes on the render thread when flaring metals
+	static class IronSteelLinesThread implements Runnable {
+		private static Thread t;
+		private static final Lock lock = new ReentrantLock();
+		private static ScanResult scanResult = new ScanResult();
+		private static int scanRange = 0;
+		private static boolean isStopping = false;
+
+		public ScanResult requestScanResult()
+		{
+			lock.lock();
+			try
+			{
+				return scanResult;
+			}
+			catch (Exception e)
+			{
+				e.printStackTrace();
+				lock.unlock();
+				return new ScanResult();	// empty ScanResult so it doesn't crash
+			}
+		}
+
+		public void releaseScanResult()
+		{
+			lock.unlock();
+		}
+
+		public void setScanRange(int range)
+		{
+			scanRange = range;
+		}
+
+		public void start()
+		{
+			if (t == null)
+			{
+				CosmereAPI.logger.info("Starting lines thread");
+				t = new Thread(this, "lines_thread");
+				t.start();
+			}
+		}
+
+		public void stop()
+		{
+			if (t != null && !isStopping)
+			{
+				isStopping = true;
+			}
+		}
+
+		private void setScanResult(ScanResult result)
+		{
+			scanResult = result;
+		}
+
+		@Override
+		@OnlyIn(Dist.CLIENT)
+		public void run() {
+			while (!isStopping) {
+				final Minecraft mc = Minecraft.getInstance();
+				final ProfilerFiller profiler = mc.getProfiler();
+				ScanResult nextScan;
+				LocalPlayer playerEntity = mc.player;
+				//only update box list every so often
+				if (playerEntity.tickCount % 15 != 0)
```

Comment:
I'm less familiar with threads, but would doing something like this be helpful?
`(playerEntity.tickCount + 5) % 15`, where 5 is whatever we feel like.

It's something I've been thinking adding elsewhere, which would help offload processing for different things onto different ticks.

If that doesn't matter, since this is already happening on a different thread, then feel free to ignore 😄

Reply:
I'm honestly not sure. Since it's a different thread, we might be able to remove the restriction entirely, but also, that might be too performance heavy? I decided to leave it up to you. I'll try the `playerEntity.tickCount + 5` and see if it makes a difference

Reply:
Tried playing with it. Either I have a super computer, or removing the restriction did almost nothing to performance. It was jarring to see highlights popping in and out though, so some restriction is probably necessary. I set it to `(playerEntity.tickCount + 5) % 5` for now, still a little jarring, but much better. I'd love to hear your thoughts once I've pushed.

Reply:
Maybe we should just remove it outright and see if anyone complains 👀 I have some peeps on the discord server with potato computers that we can get to test

Reply:
Would be interesting to see, at least. Worst case we can just hop on back to once per 15 ticks, just like before

Reply:
I'll get you to make that change, and then will make some builds off the branch and send a call out to the discord peeps 😎

Reply:
Done! Ready for the tater testers! I wish their PCs luck o7